### PR TITLE
Feature/add features to ae address

### PR DESCRIPTION
**Pull request by sadiqevani**

### CODE REVIEW

File: src/components/ae-address/ae-address.vue [hunk 1]
```diff
@@ -86,5 +95,31 @@ export default {
   > li {
     list-style: none;
   }
+
+  &.short {
+    grid-template-columns: 1fr 1fr 1fr 1fr 1fr;
```
Comment (davidyuk) [inline src/components/ae-address/ae-address.vue]:
```suggestion
    grid-template-columns: repeat(5, 1fr);
```

File: src/directives/copy-to-clipboard.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import copy from 'clipboard-copy';
+
+/**
+ * The following directive attaches an event listener
+ * to the element, and copies the value provided
+ * to the directive
+ * @return {Object}
+ */
+export default {
+  inserted: (el, binding) => el.addEventListener('click', () => copy(binding.value).then(() => {
```
Comment (davidyuk) [inline src/directives/copy-to-clipboard.js]: Why don't use `await` here?
```suggestion
  inserted: (el, binding) => el.addEventListener('click', async () => {
    await copy(binding.value);
```

File: src/directives/copy-to-clipboard.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import copy from 'clipboard-copy';
+
+/**
+ * The following directive attaches an event listener
+ * to the element, and copies the value provided
+ * to the directive
```
Comment (davidyuk) [inline src/directives/copy-to-clipboard.js]: You have missed `v-copied-to-clipboard` class description here.

File: src/components/ae-address/ae-address.vue [hunk 4]
```diff
@@ -1,5 +1,12 @@
 <template>
-  <ul class="ae-address" :style="{ gridGap: gap }" v-remove-spaces-on-copy>
+  <ul
+    ref="address"
+    class="ae-address"
+    :class="[ length, { copied } ]"
```
Comment (davidyuk) [inline src/components/ae-address/ae-address.vue]: `copied` is not defined anywhere.

**Review by davidyuk — COMMENTED** [review]
according to which designs the “address copied” label is implemented?

File: src/directives/copy-to-clipboard.js (hunk 2 above)
Comment (sadiqevani) [reply src/directives/copy-to-clipboard.js]: @davidyuk doesn't work.

File: src/components/ae-address/ae-address.vue (hunk 1 above)
Comment (sadiqevani) [reply src/components/ae-address/ae-address.vue]: updated

File: src/directives/copy-to-clipboard.js (hunk 3 above)
Comment (sadiqevani) [reply src/directives/copy-to-clipboard.js]: updated

File: src/components/ae-address/ae-address.vue (hunk 4 above)
Comment (sadiqevani) [reply src/components/ae-address/ae-address.vue]: removed

**Comment by sadiqevani** [comment]
> according to which designs the “address copied” label is implemented?

There was a discussion with Ray and Emil, and decided to implement this feature. As it's needed on token-migration.

**Comment by sadiqevani** [comment]
@davidyuk 

Also another thing we need to do, is to have the copy functionality dynamic, so we can enable/disable it, any issues/suggestions on how to handle this with a directive?

**Review by davidyuk — APPROVED** [review]

**Comment by davidyuk** [comment]
@sadiqevani As a workaround, I can propose to disable `copyToClipboard` directive if the passed value is empty.

File: src/directives/copy-to-clipboard.js (hunk 2 above)
Comment (davidyuk) [reply src/directives/copy-to-clipboard.js]: Why you have set `regenerator` filed to false in `.babelrc`?
4a3f2c8e81504975d8fda56a285edabdfe794994

Comment (sadiqevani) [reply src/directives/copy-to-clipboard.js]: @davidyuk I don't remember the reason, though thanks for the update. I'll merge the branch now.

**Comment by sadiqevani** [comment]
> As a workaround, I can propose to disable copyToClipboard directive if the passed value is empty.

But the value passed is directly taken from the address value, should we have a secondary value for copying to clip board?

**Pull request merged**